### ISSUE TITLE
Release generation should initialize the ability to create a release from the target project

### ISSUE THREAD

**Issue by choptastic** [nitrogen/nitrogen#58]
Right now, the standard Nitrogen setup creates a release which includes Nitrogen code.  This is great for development, but when it comes time for upgrading, it'd be nice to be able to take advantage of the actual auto-release upgrade system built into Erlang.

This is where a generatable release comes in.

Essentially, this would set up a rel directory with a default reltool.config file. Then include a `make release` rule for the Makefile, and in the end it would create an actual release for the project to do a more seamless hot upgrade than relying on `sync`

In short: from the nitrogen repo:

`make rel_whatever` creates an initial development environment for working, then from that "development release", `make release` would create the actual production-level release for the destination machine.
